ngx-fudis/projects/ngx-fudis/src/lib/components/button/button.component.spec.ts

```diff
@@ -9,113 +16,130 @@ describe('ButtonComponent', () => {
+      fudisIconRotateArray.forEach((rotate) => {
+        fixture.componentRef.setInput('iconRotate', `${rotate}`);
+        fixture.detectChanges();
+
+        const iconClasses = getButton().querySelector('svg')?.classList;
+
+        expect(iconClasses).toContain(`fudis-icon__rotate__${rotate}`);
+      });
     });
   });
 
-  describe('button clicked', () => {
-    it('should emit events when the button is enabled', () => {
-      let clicked = false;
-      component.label = 'Testing clicking';
-      component.handleClick.subscribe(() => {
-        clicked = true;
-      });
+  describe('Event emitters', () => {
+    it('should emit when button is clicked', () => {
+      jest.spyOn(component.handleClick, 'emit');
 
-      getButton()?.click();
+      getButton().click();
 
-      expect(clicked).toEqual(true);
+      expect(component.handleClick.emit).toHaveBeenCalled();
     });
 
-    it('should not emit events when the button is disabled', () => {
-      let clicked = false;
-      component.handleClick.subscribe(() => {
-        clicked = true;
-      });
+    it('should emit when button is focused', () => {
+      jest.spyOn(component.handleFocus, 'emit');
 
+      getButton().focus();
+
+      expect(component.handleFocus.emit).toHaveBeenCalled();
+    });
+
+    it('should emit when button is blurred', () => {
+      jest.spyOn(component.handleBlur, 'emit');
+
+      getButton().focus();
+      getButton().blur();
+
+      expect(component.handleBlur.emit).toHaveBeenCalled();
+    });
+
+    it('should not emit when button is disabled', () => {
+      jest.spyOn(component.handleClick, 'emit');
       component.disabled = true;
-      component.label = 'Testing disabled state';
       fixture.detectChanges();
 
-      getButton()?.click();
+      getButton().click();
 
-      expect(clicked).toEqual(false);
+      expect(component.handleClick.emit).not.toHaveBeenCalled();
     });
   });
 
-  describe('button with icon, aria-label and label hidden', () => {
-    it('should have icon component present if icon and label and aria-label has been given as an Input', () => {
+  describe('Button as MenuButton', () => {
+    it('should toggle dropdown menu', () => {
+      jest.spyOn(component, 'toggleMenu');
+
       fixture.componentRef.setInput('icon', 'three-dots');
       fixture.componentRef.setInput('label', 'Open additional menu');
       fixture.componentRef.setInput('labelHidden', true);
       fixture.componentRef.setInput('ariaLabel', 'It has nice things to click');
-      fixture.componentRef.setInput('type', 'button');
-
+      fixture.componentRef.setInput('asMenuButton', true);
       fixture.detectChanges();
 
-      expect(getButton().getAttribute('aria-label')).toBeTruthy();
+      getButton().click();
+
+      expect(component.toggleMenu).toHaveBeenCalled();
+
+      expect(!!getButton().getAttribute('aria-expanded')).toEqual(true);
```

Comment:
This failes if expected to be false before toggling

Reply:
Fixed